File: SparkStudy/src/main/scala/com/project/DAO/categoryClickCountDAO.scala
package com.project.DAO

import com.project.domain.categoryClickCount
import com.study.utils.HBaseUtils
import org.apache.hadoop.hbase.client.Get
import org.apache.hadoop.hbase.util.Bytes

import scala.collection.mutable.ListBuffer

object categoryClickCountDAO {

  val tableName = "click_count"
  val cf = "info"
  val qualifier = "click_count"

  //保存数据
  def save(list:ListBuffer[categoryClickCount]): Unit ={
      val table = HBaseUtils.getInstance().getTable(tableName)
      for(els<-list){
        table.incrementColumnValue(Bytes.toBytes(els.categoryID),Bytes.toBytes(cf),Bytes.toBytes(qualifier),els.clickCount)
      }
  }

  def count(day_category:String):Long= {
    val table = HBaseUtils.getInstance().getTable(tableName)
    val get = new Get(Bytes.toBytes(day_category))
    val value = table.get(get).getValue(Bytes.toBytes(cf),Bytes.toBytes(qualifier))
    if(value == null){
      0L
    }else {
      Bytes.toLong(value)
    }
  }

  def main(args: Array[String]): Unit = {
    val list = new ListBuffer[categoryClickCount]
    list.append(categoryClickCount("20171122_8",300))
    save(list)
    print(count("20171122_8"))
  }
}

File: spark_web/src/main/java/com/study/dao/CategoryClickCountDAO.java
package com.study.dao;

import com.study.domain.CategoryClickCount;
import com.study.utils.HBaseUtils;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Component
public class CategoryClickCountDAO {
    public List<CategoryClickCount> query(String day) throws IOException {
        List<CategoryClickCount> list = new ArrayList<>();
        Map<String,Long> map = HBaseUtils.getInstance().query("category_clickcount",day);
        for(Map.Entry<String,Long> entry:map.entrySet()){
            CategoryClickCount model = new CategoryClickCount();
            model.setName(entry.getKey());
            model.setValue(entry.getValue());

            list.add(model);
        }
        return list;
    }
}
